src/components/Recipes.js

```diff
@@ -0,0 +1,146 @@
+import { v4 as uuid } from 'uuid';
+import List from '@mui/material/List';
+import Divider from '@mui/material/Divider';
+import Typography from '@mui/material/Typography';
+import React, { useEffect, useState } from 'react';
+import * as recipesClient from '../api/recipes';
+import AddRecipe from './AddRecipe';
+import Recipe from './Recipe';
+
+function Recipes() {
+  const [recipes, setRecipes] = useState([]);
+
+  const fetchRecipes = () => {
+    recipesClient
+      .getRecipes()
+      .then(setRecipes);
+  };
+
+  useEffect(fetchRecipes, []);
+
+  const addRecipe = ({ name, description, ingredients = [] }) => {
+    recipesClient
+      .addRecipe({ name, description, ingredients })
+      .then((newRecipe) => {
+        setRecipes([...recipes, newRecipe]);
+      });
+  };
+
+  const editRecipe = ({ recipeId, name }) => {
+    recipesClient
+      .editRecipe({ recipeId, name })
+      .then(fetchRecipes);
+  };
+
+  const removeRecipe = ({ recipeId }) => {
+    recipesClient
+      .removeRecipe({ recipeId })
+      .then(fetchRecipes);
+  }
+
+  const removeIngredient = ({ recipeId, ingredientId }) => {
+    const recipe = recipes.find(r => r.id === recipeId);
+    const newIngredients = recipe.ingredients.filter(i => i.id !== ingredientId);
+
+    recipesClient
+      .editRecipe({
+        recipeId,
+        ingredients: newIngredients,
+      })
+      .then(fetchRecipes);
+  };
+
+  const editIngredient = ({ recipeId, ingredientId, name }) => {
+    const recipe = recipes.find(r => r.id === recipeId);
+
+    const ingredients = recipe.ingredients.map(ingredient => {
+      if (ingredient.id === ingredientId) {
+        return {
+          name,
+        };
+      }
+
+      return ingredient;
+    });
+
+    recipesClient
+      .editRecipe({
+        recipeId,
+        ingredients,
+      })
+      .then(fetchRecipes);
+
+    const newRecipies = recipes.map(recipe => {
+      if (recipe.id === recipeId) {
+        const newIngredients = recipe.ingredients.map(ingredient => {
+          if (ingredient.id === ingredientId) {
+            return {
+              ...ingredient,
+              name,
+            }
+          }
+
+          return ingredient;
+        });
+
+        return {
+          ...recipe,
+          ingredients: newIngredients,
+        };
+      }
+
+      return recipe;
+    });
+
+    setRecipes(newRecipies);
+  };
+
+  const editRecipeDescription = ({ recipeId, newDescription }) => {
```

Comment:
Does this function edit the recipe or only the state?

Reply:
It modifies the state. I forgot about updating this function to make it modify the recipe in the backend. Just updated it. Good catch!